src/validateRegisterForm.test.js

```diff
@@ -26,5 +26,204 @@ describe(`Function 'validateRegisterForm':`, () => {
+      const isValid = validateRegisterForm('test@mail.com', 'P@ssword1!Long');
+
+      expect(isValid.code).toBe(200);
+      expect(isValid.message).toBe('Email and password are valid.');
+    });
+
+    it(`should return error for pass with more than 16 characters`, () => {
+      const invalidPassword = validateRegisterForm(
+        'test@mail.com', 'P@ssword1!LongLong');
+
+      expect(invalidPassword.code).toBe(422);
+      expect(invalidPassword.message).toBe('Password is invalid.');
+    });
+
+    it(`should return success valid passcontaining at least 1 digit`, () => {
+      const isValid = validateRegisterForm('test@mail.com', 'P@ssword1!');
+
+      expect(isValid.code).toBe(200);
+      expect(isValid.message).toBe('Email and password are valid.');
+    });
+
+    it(`should return error for a password without any digits`, () => {
+      const invalidPassword = validateRegisterForm(
+        'test@mail.com', 'P@ssword!');
+
+      expect(invalidPassword.code).toBe(422);
+      expect(invalidPassword.message).toBe('Password is invalid.');
+    });
+  });
+
+  describe(`Requirements for the valid email:`, () => {
+    it(`should return success for a valid email with English letters`, () => {
+      const isValid = validateRegisterForm('test@mail.com', 'P@ssword1!');
+
+      expect(isValid.code).toBe(200);
+      expect(isValid.message).toBe('Email and password are valid.');
+    });
+
+    it(`should return error for an email without English letters`, () => {
+      const invalidEmail = validateRegisterForm(
+        'Илья@mail.com', 'P@ssword1!');
+
+      expect(invalidEmail.code).toBe(422);
+      expect(invalidEmail.message).toBe('Email is invalid.');
+    });
+
+    it(`should return success for a valid email with digits`, () => {
+      const isValid = validateRegisterForm('test123@mail.com', 'P@ssword1!');
+
+      expect(isValid.code).toBe(200);
+      expect(isValid.message).toBe('Email and password are valid.');
+    });
+
+    it(`should return error for an email without digits`, () => {
+      const invalidEmail = validateRegisterForm('test@com', 'P@ssword1!');
+
+      expect(invalidEmail.code).toBe(422);
+      expect(invalidEmail.message).toBe('Email is invalid.');
+    });
+
+    it('should return success for valid email with special characters', () => {
+      const isValid = validateRegisterForm('test123@mail.com', 'P@ssword1!');
+
+      expect(isValid.code).toBe(200);
+      expect(isValid.message).toBe('Email and password are valid.');
+    });
+
+    it(`should return error for email with invalid special characters`, () => {
+      const invalidEmail = validateRegisterForm('test@!mail.com', 'P@ssword1!');
+
+      expect(invalidEmail.code).toBe(422);
+      expect(invalidEmail.message).toBe('Email is invalid.');
+    });
+
+    it(`should return success for a valid email with periods`, () => {
+      const isValid = validateRegisterForm('test.mail@mail.com', 'P@ssword1!');
+
```

Comment:
Let`s clarify this description because you have a negative check below "email with double dots"